python/pyspark/sql/datasource.py

```diff
@@ -469,6 +501,200 @@ def stop(self) -> None:
         ...
 
 
+class SimpleInputPartition(InputPartition):
+    def __init__(self, start: dict, end: dict):
+        self.start = start
+        self.end = end
+
+
+class PrefetchedCacheEntry(InputPartition):
```

Comment:
Ditto why do we need this in the public API?

Reply:
Where should we move these functions to if we don't want them to be public API?